src/runtime/NewReference.cs

```diff
@@ -27,14 +27,17 @@ public PyObject MoveToPyObject()
             this.pointer = IntPtr.Zero;
             return result;
         }
+
         /// <summary>
         /// Removes this reference to a Python object, and sets it to <c>null</c>.
         /// </summary>
         public void Dispose()
         {
-            if (!this.IsNull())
-                Runtime.XDecref(this.pointer);
-            this.pointer = IntPtr.Zero;
+            if (this.IsNull())
+            {
+                return;
+            }
+            Runtime.XDecref(this.pointer);
```

Comment:
To avoid accidental double `XDecref`, it is necessary to null the `pointer` here.